app/Repositories/WordRepository.php

```diff
@@ -89,8 +89,8 @@ public function firstOrNew(string $literal, Language $language, $user): Word
         ]);
     }
 
-    public function getWordWithRequesters($wordId)
+    public function getWordWithRequesters(Word $word): Word
     {
-        return Word::withCount('requesters')->find($wordId);
+        return Word::withCount('requesters')->find($word->id)->first();
```

Comment:
Hvad er forskellen?
Og har du taget højde for at den kan komme lige efter man har nedvotet en word til nul request, og word'en dermed slettes? Denne metode er 100% sikker på at altid returerer en Word. Det er så ikke muligt at returner en null.

Reply:
Det kommer an på om vil du modtage en eloquent eller en Word-model. Jeg fjerner bare den.

Reply:
Bare en string... en id. Det er også meget fint i denne tilfælde. Så kan de andre steder også kalde det med id alene, og det vil så virke meget mere generelt end at fastelåse os til at skal have word. Så skal jeg lave `$word = Word::find($wordId)` før jeg kan kalde `WordRepository->($word)` - hvis jeg kun har en $wordId at råde over.

Reply:
Metoden bliver kun brugt ifm toggleRequest, som formålet med denne issue at ændre fra wordId til literal. Så jeg forstår ikke hvorfor du stadig forventer at API'en spytter stadig med wordId?

Reply:
Jo - problemet er blot at du pt. er ind i `WordRepository` - som er forventet at skulle kunne håndtere på tværs af en masse af services og controllers. I lige netop det tilfælde med toogleRequest, har vi allerede en Word model som vi kan anvende til denne metode. MEN der kunne jo være i fremtiden at der dukker en anden behov op, der også har brug for at fetche en word med requesters - og hvis den anden metode kun har en wordId at råde over, så ville det være dumt at omgøre det hele, når vi i denne metode allerede kan "begrænse" os til at kun modtage WordId, da det er den eneste variable vi har brug for, før at løse lige netop den opgave som metoden gør... Metoden har ikke brug for en word model - blot en id over Word...